.github/workflows/build_linux.yaml

```diff
@@ -38,6 +38,9 @@ jobs:
             - name: Install missing tools
               if: runner.environment == 'self-hosted'
               run: |
+                  # Sanity check that the arch is arm64 as we expect
+                  [[ $(uname -p) == "aarch64" ]] || exit 1
+
```

Comment:
Does this mean that the conditional in the runs-on above is redundant if we always run on aarch64?

Reply:
> Does this mean that the conditional in the runs-on above is redundant if we always run on aarch64?

We don't. This workflow runs on both amd64 & aarch64
For the latter we need to use `dind-l-arm64` which is a self-hosted runner. This runner has historically sometimes been `amd64` due to a bug in the K8S config. This check compares the real arch to the arch we expect

Reply:
oh right, so they are both arm64, just different types

Reply:
@dbkr no not quite

amd64 (i.e x64) and aarch64 (i.e arm64)

We run this workflow for both, but we only use self-hosted runners for aarch64 as those are not available as free runners in Github at this time (except M1 macos runners)

So this run section "Install missing tools" is a set of workarounds for our self-hosted runners, including this sanity check. We *only* use self-hosted runners for aarch64/arm64. Not x64/amd64.